### PR TITLE
Flow extensions as config extensions

### PR DESCRIPTION
**Pull request by isaacroldan**
### WHY are these changes introduced?
Initially Flow extensions were defined as ui-extensions, but that is no longer necessary since we introduced the concept of generic extensions.

### WHAT is this pull request doing?
- Define flow extension with the new ExtensionSpecification.
- Add flow extensions templates

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/app/src/cli/models/templates/ui-specifications/flow_trigger_definition.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  types: [
+    {
+      url: 'https://github.com/Shopify/cli',
+      type: 'flow_trigger_definition',
+      extensionPoints: [],
+      supportedFlavors: [
+        {
+          name: 'Config only',
+          value: 'config-only',
+          path: 'templates/ui-extensions/projects/flow_trigger_definition',
```
[the remark this answers is not in the document]
Comment (isaacroldan) [reply packages/app/src/cli/models/templates/ui-specifications/flow_trigger_definition.ts:20]: yeah totally, you can put it anywhere inside the `templates` folder

[the remark this answers is not in the document]
Comment (MaxLeduc) [reply packages/app/src/cli/models/templates/ui-specifications/flow_trigger_definition.ts]: Thank you!

File: packages/app/src/cli/models/templates/ui-specifications/flow_action_definition.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import {ExtensionTemplate} from '../../app/template.js'
```
Comment (MaxLeduc) [inline packages/app/src/cli/models/templates/ui-specifications/flow_action_definition.ts:1]: Quick question: do these need to be located in the `ui-specifications` folder and if yes will that change in the future or is this how generic extensions specs will always be created?

Comment (isaacroldan) [reply packages/app/src/cli/models/templates/ui-specifications/flow_action_definition.ts:1]: no, no need to have it here, forgot to move these, we can move them outside this folder.
Not that it's relevant anyway, is just for us to organize code internally

Comment (MaxLeduc) [reply packages/app/src/cli/models/templates/ui-specifications/flow_action_definition.ts:1]: Ah I see the import below. Any objections to moving this to a `config-only-specifications` or `generic-specifications` folder?

Comment (MaxLeduc) [reply packages/app/src/cli/models/templates/ui-specifications/flow_action_definition.ts:1]: Sounds good, thank you!

**Comment by MaxLeduc** [comment]
Closing in favour of https://github.com/Shopify/cli/pull/2106

**Pull request closed without merging**